### PR TITLE
Define schema for kubeAuth stanza supporting k8s 1.6 (part of #502).

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

File: schemas/config/v1/kubeAuthAuthnEntry.json [hunk 1]
```diff
@@ -6,17 +6,19 @@
...
     "basic": {
-      "description": "Users to be created",
+      "description": "List of user/password to be created.",
       "items": { "$ref": "kubeAuthBasicUser.json" },
       "type": "array"
     },
     "default_basic_user": {
-      "description": "Default user for kubeauth file",
+      "description": "Default user for kubeauth file.",
       "type": "string"
```
Comment (mikeln) [inline schemas/config/v1/kubeAuthAuthnEntry.json]: So, this value `default_basic_user` really has to be one of the `basic[].user` values.  Not a blocking issue, but something to start to think about, how to handle that type of requirement.

File: schemas/config/v1/kubeAuthAuthzEntry.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      "description": "Kubernetes RBAC configuration.",
+      "properties": {
+        "super_user": {
+          "description": "Determines the authorization-rbac-super-user flag used to start the apiserver in versions of kubernetes <= 1.5",
+          "type": "string",
+          "default": "placeholder"
+        }
+      },
+      "required": [
+        "super_user"
```
Comment (mikeln) [inline schemas/config/v1/kubeAuthAuthzEntry.json]: This is only required pre1.6.  I guess it doesn't hurt in 1.6.  
How does required work exactly?   
My concern: if rbac is off, then authz needs to be: `authz: {}`.  Does this required break that?

File: schemas/config/v1/kubeAuthBasicUser.json [hunk 3]
```diff
@@ -6,11 +6,15 @@
...
     },
     "password": {
-      "description": "The password for the user account",
+      "description": "The password for the user account.",
+      "type": "string"
+    },
+    "group": {
+      "description": "The groups for the user account.",
       "type": "string"
     }
```
Comment (mikeln) [inline schemas/config/v1/kubeAuthBasicUser.json]: `group` is only needed for 1.6+.   Also, for this entry, `user`,`password` are required for all versions, `group` is also required for 1.6.

File: schemas/config/v1/kubeAuthAuthnEntry.json (hunk 1 above)
Comment (davidewatson) [inline schemas/config/v1/kubeAuthAuthnEntry.json]: Draft 5 may help with this, but for now we must check in ansible.

File: schemas/config/v1/kubeAuthBasicUser.json (hunk 3 above)
Comment (davidewatson) [inline schemas/config/v1/kubeAuthBasicUser.json]: Without creating a new kind, we can't make group required some of the time. If there end up being many of these corner cases we might want to do something different than check in ansible.

File: schemas/config/v1/kubeAuthAuthzEntry.json (hunk 2 above)
Comment (davidewatson) [inline schemas/config/v1/kubeAuthAuthzEntry.json]: Good call. I'll make super_user optional.

**Comment by davidewatson** [comment]
Passed cluster creation for both AWS and GKE: https://common-jenkins.kubeme.io/job/K2/job/PR-511/4/console

**Comment by davidewatson** [comment]
AWS cluster create passed: https://common-jenkins.kubeme.io/job/K2/job/PR-512/

**Pull request merged**